### PR TITLE
Added libuast's iterators.

### PR DESCRIPTION
**Pull request by juanjux**
Signed-off-by: Juanjo Alvarez <juanjo@sourced.tech>

### CODE REVIEW

File: tools/bindings.go [hunk 1]
```diff
@@ -10,14 +10,32 @@ import (
...
 
+// Traversal strategy for UAST trees
+type TreeOrder int
+const (
+	// PreOrder traversal
+	PreOrder TreeOrder = iota
+	// PostOrder traversal
+	PostOrder
+	// LevelOrder (aka breadth-first) traversal
+	LevelOrder
```
Comment (vmarkovtsev) [inline tools/bindings.go]: Would be nice to have ASCII tree images which describe the orders.

Also, not sure which order corresponds to the natural, source code occurrence?

**Comment by vmarkovtsev** [comment]
Why do we lock a mutex and stop GC for every function btw?

**Comment by juanjux** [comment]
Neither currently, but as talked before, it should be possible doing a first iteration in any order and inserting sorted by line numbers in a list. I'll open an issue for libuast.

**Comment by juanjux** [comment]
Because currently libuast is not thread-safe. The GC to avoid Go deleting the nodes once there aren't more references in Go code (but the iterators could have references in their internal data structures that are not tracked by Go GC since they're C code), which would invalidate libuast pointers. They'll be freed when you dispose() the iterator.

**Pull request merged**